warp10/src/main/java/io/warp10/continuum/gts/GTSHelper.java

```diff
@@ -1897,14 +1897,65 @@ public static final GeoTimeSerie subSerie(GeoTimeSerie gts, long starttimestamp,
     // Extract values/locations/elevations that lie in the requested interval
     //
 
-    // We know how many data will the new GTS so we provision arrays to receive the data.
+    // We know how many data the new GTS will have so we provision arrays to receive the data.
     int count = lastidx - firstidx + 1;
+
     GTSHelper.multiProvision(subgts, gts.type, count, count);
```

Comment:
In multiProvision if subgts.type is UNDEFINED then the value array will be allocated according to gts.type so allocation will be correctly done. Did you see another execution path which proves otherwise?

Reply:
Ok got your execution path, the case when gts.ticks is not null but type is UNDEFINED.